python/mxnet/context.py

```diff
@@ -60,18 +62,16 @@ class Context(object):
     >>> gpu_array.context
     gpu(1)
     """
-    # static class variable
-    default_ctx = None
```

Comment:
This will break a lot of existing code. Is this change intended for 2.0?

Reply:
I don't think this was meant to be public API. current_context() is used for accessing this.

Reply:
@piiswrong told me that default_ctx, devtype2str and devstr2type where not meant to be public API. This will also be renamed to _default_ctx. Users are expected to use `current_context` or `default_context` in test_utils. I agree though that this will break some existing code. I was of the opinion that the release after 1.2 is 2.0 but I am not sure.

Reply:
If there are users who depend on this, we can make a backward compatible wrapper with classmethod and property:https://stackoverflow.com/questions/1697501/python-staticmethod-with-property

Reply:
```
_default_ctx = threading.local()
@property
@classmethod
def default_ctx():
   return _default_ctx.value
```